pkg/account/storage/v2/api_key.go

```diff
@@ -314,8 +314,12 @@ func (s *accountStorage) ListAPIKeys(
 	}
 	nextOffset := offset + len(apiKeys)
 	var totalCount int64
-	countQuery, countWhereArgs := mysql.ConstructQueryAndWhereArgs(selectAPIKeyV2CountSQLQuery, options)
-	err = s.qe.QueryRowContext(ctx, countQuery, countWhereArgs...).Scan(&totalCount)
+	if options != nil {
+		whereQuery, countWhereArgs := mysql.ConstructWhereSQLString(options.CreateWhereParts())
+		err = s.qe.QueryRowContext(ctx, selectAPIKeyV2CountSQLQuery+whereQuery, countWhereArgs...).Scan(&totalCount)
+	} else {
+		err = s.qe.QueryRowContext(ctx, selectAPIKeyV2CountSQLQuery).Scan(&totalCount)
+	}
```

Comment:
WDYT of creating a `ConstructWhereSQLString` helper in the query.go to reduce duplication logic?

```suggestion
	countQuery, countArgs := mysql.ConstructCountQuery(countAccountsV2SQL, options)
	err = s.qe.QueryRowContext(ctx, countQuery, countArgs...).Scan(&totalCount)
```
---

`mysql/query.go`
```go
// ConstructCountQuery builds a count query with optional filtering.
// It takes a base count SQL query and ListOptions, and returns the complete query and its arguments.
// The base SQL should be a valid COUNT query without WHERE clause.
func ConstructCountQuery(baseSQL string, options *ListOptions) (query string, args []interface{}) {
	if options != nil {
		whereQuery, countWhereArgs := ConstructWhereSQLString(options.CreateWhereParts())
		return baseSQL + whereQuery, countWhereArgs
	}
	return baseSQL, nil
}
```

Reply:
Added `ConstructQueryAndWhereArgsOnlyUseWherePart()` and consolidated common code.
https://github.com/bucketeer-io/bucketeer/pull/1781/commits/0a94775d255f55041e38165421777980d3c08003